tests/e2e/utils/service_utils.go

```diff
@@ -159,59 +169,24 @@ func WaitServiceExposure(cs clientset.Interface, namespace string, name string,
 	return service, nil
 }
 
-func isInternalService(service *v1.Service) bool {
-	var (
-		val string
-		ok  bool
-	)
-	if val, ok = service.Annotations[consts.ServiceAnnotationLoadBalancerInternal]; !ok {
-		return false
-	}
-
-	return strings.EqualFold(val, "true")
-}
-
-// ValidateExternalServiceConnectivity validates the connectivity of the public service IP
-func ValidateExternalServiceConnectivity(serviceIP string, port int) error {
-	// the default nginx port is 80, skip other ports
-	if port != 80 {
-		return nil
+// ValidateServiceConnectivity validates the connectivity of the internal Service IP
+func ValidateServiceConnectivity(ns, execPod, serviceIP string, port int, protocol v1.Protocol) error {
+	udpOption := ""
+	if protocol == v1.ProtocolUDP {
+		udpOption = "-u"
 	}
-
-	err := wait.PollImmediate(pullInterval, pullTimeout, func() (done bool, err error) {
-		resp, err := http.Get(fmt.Sprintf("http://%s:%d", serviceIP, port))
-		if err != nil {
-			Logf("got error %v, will retry", err)
-			return false, nil
-		}
-
-		if 200 <= resp.StatusCode && resp.StatusCode < 300 {
-			Logf("succeeded")
-			return true, nil
-		}
-
-		Logf("got status code %d", resp.StatusCode)
-		return false, nil
-	})
-
-	Logf("validation finished")
-	return err
-}
-
-// ValidateInternalServiceConnectivity validates the connectivity of the internal Service IP
-func ValidateInternalServiceConnectivity(ns, execPod, serviceIP string, port int) error {
-	cmd := fmt.Sprintf(`curl -m 5 http://%s:%d`, serviceIP, port)
-	pollErr := wait.PollImmediate(pullInterval, pullTimeout, func() (bool, error) {
+	cmd := fmt.Sprintf(`nc -vz -w 4 %s %s %d`, udpOption, serviceIP, port)
+	pollErr := wait.PollImmediate(pullInterval, 3*pullTimeout, func() (bool, error) {
```

Comment:
If we add pod health probe configs in manifests, We can validate pod status instead.

Reply:
I think we are checking service connectivity here so nc to service ip makes more sense? Could you please share an example for reference in e2e?